### PR TITLE
Dynamically reduce dartdoc timeout with total time budget option.

### PR DESCRIPTION
**Pull request by isoos**
Should address https://github.com/dart-lang/pub-dev/issues/7430

### CODE REVIEW

File: lib/src/package_analyzer.dart [hunk 1]
```diff
@@ -28,12 +28,14 @@ import 'utils.dart';
...
   final String? pubHostedUrl;
   final String? dartdocOutputDir;
+  final Duration? totalTimeBudget;
   final Duration? dartdocTimeout;
   final int? lineLength;
 
   InspectOptions({
     this.pubHostedUrl,
     this.dartdocOutputDir,
+    this.totalTimeBudget,
```
Comment (sigurdm) [inline lib/src/package_analyzer.dart:38]: Could we document this a bit.
Can you pass both timeouts? How do they interact? Are there defaults?

Comment (isoos) [reply lib/src/package_analyzer.dart:54]: Updated this and other options. PTAL

File: lib/src/package_analyzer.dart [hunk 2]
```diff
@@ -26,15 +26,33 @@ import 'tool/run_constrained.dart';
 import 'utils.dart';
 
 class InspectOptions {
+  /// The PUB_HOSTED_URL to use for the package download and dependency analysis.
   final String? pubHostedUrl;
+
+  /// The output directory to copy the generated docs. When not specified,
+  /// the documents will be discarded.
```
Comment (sigurdm) [inline lib/src/package_analyzer.dart:33]:
```suggestion
  /// the generated docs will be discarded.
```

File: lib/src/package_analyzer.dart [hunk 3]
```diff
@@ -26,15 +26,33 @@ import 'tool/run_constrained.dart';
...
   final String? dartdocOutputDir;
-  final Duration? dartdocTimeout;
+
+  /// The total time budget allocated for the full analysis. `pana` may not be
+  /// able to finish the analysis withing this time, but some parts will be
```
Comment (sigurdm) [inline lib/src/package_analyzer.dart:37]:
```suggestion
  /// able to finish the analysis within this time, but some parts will be
```

**Review by sigurdm — APPROVED** [review]

**Pull request merged**